libbeat/monitoring/report/elasticsearch/client.go

```diff
@@ -152,3 +148,88 @@ func (c *publishClient) Test(d testing.Driver) {
 func (c *publishClient) String() string {
 	return "publish(" + c.es.String() + ")"
 }
+
+func (c *publishClient) publishWithXPackMonitoringBulk(params map[string]string, event publisher.Event, t interface{}) error {
```

Comment:
I'd feel a little more comfortable we `t` would be a string. nit: naming (e.g. `doctype` or `typ`)